packages/cli/src/Format.ts

```diff
@@ -54,15 +54,32 @@ Or specify a Prisma schema path
 
     const schemaPath = await getSchemaPathAndPrint(args['--schema'])
 
-    let output = await formatSchema({
-      schemaPath,
-    })
+    let output: string | undefined
+
+    try {
+      output = await formatSchema({
+        schemaPath,
+      })
+    } catch (err) {
+      if (err.exitCode === 101 || err.stderr?.includes('panicked at')) {
+        throw new RustPanic(
+          /* message */ err.shortMessage,
+          /* rustStack */ err.stack,
+          /* request */ 'format',
+          ErrorArea.INTROSPECTION_CLI, // TODO: change to FMT_CLI
+          schemaPath,
+          /* schema */ undefined,
+        )
+      }
+
+      throw err
+    }
 
     await getDMMF({
       datamodel: output,
     })
```

Comment:
No, if it fails the error is just propagated to the cli error handler, which detects whether it's a Rust panic or not

Reply:
Since `formatSchema()` is only used here, that works for now, though it could be good to handle this in the `formatSchema()` function, like for getConfig & getDmmf.

Reply:
I don't think there is a test for `formatSchema()` with a panic?
We could test manually at least and optimally add a test